astore/server/auth/auth.go

```diff
@@ -132,8 +131,17 @@ func (s *Server) Token(ctx context.Context, req *auth.TokenRequest) (*auth.Token
 				Token: box.Seal(nil, []byte(authData.Cookie), &nonce, (*[32]byte)(clientPub), (*[32]byte)(s.serverPriv)),
 			}, nil
 		}
+		// if the ca signer was present, continuing with public keys
+		b, _ := pem.Decode(req.Publickey)
+		if b == nil {
+			return nil, errors.New("public key not sent on a CA enabled server")
```

Comment:
Make this block of code optional on receiving req.PublicKey. I'd like to roll out the new CA server even before clients are updated. If you make this code not return error when req.PublicKey is not provided, it is up to the client to ask for the key to be signed or not.

Reply:
Sure thing, implemented